src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final String ADD_MESSAGE_EXCEPTION = "Can't add to index ";
+    private static final String REMOVE_MESSAGE_EXCEPTION = "Can't remove to index ";
+    private static final String SET_MESSAGE_EXCEPTION = "Can't set to index ";
+    private static final String GET_MESSAGE_EXCEPTION = "Can't get to index ";
+    private static final String NO_SUCH_MESSAGE = "No such element in array list";
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] listArray;
+    private int size = 0;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.listArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexInclusiveSize(index, ADD_MESSAGE_EXCEPTION + index);
+        if (size == listArray.length) {
+            resizeListArray(index);
+        } else {
+            System.arraycopy(listArray, index, listArray, index + 1, size - index);
+        }
+        listArray[index] = value;
+        size++;
+    }
 
+    private void checkIndex(int index, String message) {
+        if (!(index >= 0 && index < size)) {
+            throw new ArrayListIndexOutOfBoundsException(message);
+        }
+    }
+
+    private void checkIndexInclusiveSize(int index, String message) {
```

Comment:
Petro was talking about not passing the messages as method parameters, this is a bad practice, instead, we should hardcode them, and instead of creating several methods for each type of message that you have got, I'd just suggest changing it to a single more general thing like `invalid index: + index`